### PR TITLE
[try] Record failed e2e tests as artifacts

### PR DESCRIPTION
**Pull request by adamziel**
This is an exploratory PR to record mp4 videos of failed e2e tests, inspired by this comment by @noahtallen:

https://github.com/WordPress/gutenberg/pull/26664#pullrequestreview-527674249

It captures png screenshots using chrome devtools protocol, stores them on disk, and uses ffmpeg to encode an mp4 video.

Concerns:
* Is there any way to make chromium expose a screencast instead of screenshots? I thought about using WebRTC, but won't work in chromium, we'd need chrome.
* If not, could those screenshots be in jpg format?
* Also, could we directly pipe the screenshots to `ffmpeg`?
* Are there any disk space limitations for artifacts this would cause us to exceed?
* Are screenshots alone good enough? Is this worth pursuing?

cc @noahtallen @noisysocks @talldan @gziolo @draganescu

### CODE REVIEW

File: packages/e2e-tests/config/setup-debug-artifacts.js [hunk 1]
```diff
@@ -52,3 +60,61 @@ function slugify( testName ) {
...
+const startVideo = async () => {
+	frames = [];
+	if ( ! session ) {
+		session = await page.target().createCDPSession();
+	}
+	await session.send( 'Page.startScreencast', {
+		format: 'png',
+		everyNthFrame: 1,
+	} );
+	session.on( 'Page.screencastFrame', ( event ) => {
```
Comment (adamziel) [inline packages/e2e-tests/config/setup-debug-artifacts.js:76]: do we need to register a new listener for every started test? do they get cleared on `stopScreencast`?

**Comment by kevin940726** [comment]
I believe cypress supports this out of the box. Not sure if it uses puppeteer though, might need to further investigate.

**Comment by noisysocks** [comment]
> I believe cypress supports this out of the box. Not sure if it uses puppeteer though, might need to further investigate.

Just noting that we used to use Cypress but moved away from it, see https://github.com/WordPress/gutenberg/pull/5618.

**Comment by adamziel** [comment]
Now that we have Playwright, achieving this goal should be easier. I'm closing this PR.

**Pull request closed without merging**